src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py

```diff
@@ -0,0 +1,1033 @@
+        attention_maps = self.attention_store.aggregate_attention(
+            from_where=("up", "down", "mid"),
+        )
+        max_attention_per_index = self._compute_max_attention_per_index(
+            attention_maps=attention_maps,
+            indices_to_alter=indices_to_alter,
+        )
+        return max_attention_per_index
+
+    @staticmethod
+    def _compute_loss(max_attention_per_index: List[torch.Tensor], return_losses: bool = False) -> torch.Tensor:
+        """Computes the attend-and-excite loss using the maximum attention value for each token."""
+        losses = [max(0, 1.0 - curr_max) for curr_max in max_attention_per_index]
+        loss = max(losses)
+        if return_losses:
+            return loss, losses
+        else:
+            return loss
+
+    @staticmethod
+    def _update_latent(latents: torch.Tensor, loss: torch.Tensor, step_size: float) -> torch.Tensor:
+        """Update the latent according to the computed loss."""
+        grad_cond = torch.autograd.grad(loss.requires_grad_(True), [latents], retain_graph=True)[0]
+        latents = latents - step_size * grad_cond
+        return latents
+
+    def _perform_iterative_refinement_step(
+        self,
+        latents: torch.Tensor,
+        indices_to_alter: List[int],
+        loss: torch.Tensor,
+        threshold: float,
+        text_embeddings: torch.Tensor,
+        text_input,
+        step_size: float,
+        t: int,
+        max_refinement_steps: int = 20,
+    ):
+        """
+        Performs the iterative latent refinement introduced in the paper. Here, we continuously update the latent code
+        according to our loss objective until the given threshold is reached for all tokens.
+        """
+        iteration = 0
+        target_loss = max(0, 1.0 - threshold)
+        while loss > target_loss:
+            iteration += 1
+
+            latents = latents.clone().detach().requires_grad_(True)
+            _ = self.unet(latents, t, encoder_hidden_states=text_embeddings[1].unsqueeze(0)).sample
+            self.unet.zero_grad()
+
+            # Get max activation value for each subject token
+            max_attention_per_index = self._aggregate_and_get_max_attention_per_token(
+                indices_to_alter=indices_to_alter,
+            )
+
+            loss, losses = self._compute_loss(max_attention_per_index, return_losses=True)
+
+            if loss != 0:
+                latents = self._update_latent(latents, loss, step_size)
+
+            try:
+                low_token = np.argmax([l.item() if type(l) != int else l for l in losses])
+            except Exception as e:
+                logger.info(e)  # catch edge case :)
+                low_token = np.argmax(losses)
+
+            low_word = self.tokenizer.decode(text_input.input_ids[0][indices_to_alter[low_token]])
+            logger.info(f"\t Try {iteration}. {low_word} has a max attention of {max_attention_per_index[low_token]}")
```

Comment:
```suggestion
```
Think we don't need to decode words in the actual pipeline - it looks like this was mostly used for debugging no?

Reply:
probably -  in that case, I can add a `#Copied from` for the `encode_prompt` method too